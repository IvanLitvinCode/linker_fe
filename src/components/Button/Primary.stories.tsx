import type { Meta, Story } from '@storybook/react';

import type { ButtonProps } from './Button';
import { Button } from './Button';

export default {
  title: 'Needl/Button/Primary',
  component: Button,
} as Meta;

const Template: Story<ButtonProps> = (args) => <Button {...args} />;

export const PrimaryLarge = Template.bind({});
PrimaryLarge.args = {
  variant: 'primary',
  size: 'large',
  label: 'Select and Continue',
  onClick: () => null,
};

export const PrimaryMedium = Template.bind({});
PrimaryMedium.args = {
  variant: 'primary',
  size: 'medium',
  label: 'Select and Continue',
  onClick: () => null,
};

export const PrimarySmall = Template.bind({});
PrimarySmall.args = {
  variant: 'primary',
  size: 'small',
  label: 'Select and Continue',
  onClick: () => null,
};

export const PrimaryMediumDisabled = Template.bind({});
PrimaryMediumDisabled.args = {
  variant: 'primary',
  size: 'medium',
  label: 'Select and Continue',
  disabled: true,
  onClick: () => null,
};

export const PrimaryMediumLoading = Template.bind({});
PrimaryMediumLoading.args = {
  variant: 'primary',
  size: 'medium',
  loading: true,
  label: 'Select and Continue',
  onClick: () => null,
};

export const PrimaryMediumFullWidth = Template.bind({});
PrimaryMediumFullWidth.args = {
  variant: 'primary',
  size: 'medium',
  fullWidth: true,
  label: 'Select and Continue',
  onClick: () => null,
};

export const StartIcon = Template.bind({});
StartIcon.args = {
  variant: 'primary',
  label: 'Button',
  startIcon: <i className='neu-share' />,
  onClick: () => null,
};

export const EndIcon = Template.bind({});
EndIcon.args = {
  variant: 'primary',
  label: 'Button',
  endIcon: <i className='neu-share' />,
  onClick: () => null,
};
