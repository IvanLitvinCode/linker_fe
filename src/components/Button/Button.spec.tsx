import { render } from '@testing-library/react';

import { Button } from './Button';

describe('Button', () => {
  it('default button', () => {
    const defaultButton = render(<Button label='Button' />);
    expect(defaultButton.container.firstChild).toMatchSnapshot();
  });

  it('default primary button', () => {
    const defaultButton = render(<Button variant='primary' label='Button' />);
    expect(defaultButton.container.firstChild).toMatchSnapshot();
  });

  it('primary and disabled button', () => {
    const defaultButton = render(
      <Button variant='primary' disabled label='Button' />
    );
    expect(defaultButton.container.firstChild).toMatchSnapshot();
  });

  it('primary and large button', () => {
    const defaultButton = render(
      <Button variant='primary' size='large' label='Button' />
    );
    expect(defaultButton.container.firstChild).toMatchSnapshot();
  });

  it('primary and small button', () => {
    const defaultButton = render(
      <Button variant='primary' size='small' label='Button' />
    );
    expect(defaultButton.container.firstChild).toMatchSnapshot();
  });

  it('default full-width button', () => {
    const defaultButton = render(<Button fullWidth label='Button' />);
    expect(defaultButton.container.firstChild).toMatchSnapshot();
  });

  it('default text button', () => {
    const defaultButton = render(<Button variant='action' label='Button' />);
    expect(defaultButton.container.firstChild).toMatchSnapshot();
  });

  it('text button with background', () => {
    const defaultButton = render(
      <Button variant='action' label='Button' className='bg-white' />
    );
    expect(defaultButton.container.firstChild).toMatchSnapshot();
  });

  it('default link button', () => {
    const defaultButton = render(<Button variant='tertiary' label='Button' />);
    expect(defaultButton.container.firstChild).toMatchSnapshot();
  });

  it('oauth button', () => {
    const defaultButton = render(
      <Button
        variant='oauth'
        label='Button'
        startIcon={<i className='neu-share' />}
      />
    );
    expect(defaultButton.container.firstChild).toMatchSnapshot();
  });
});
