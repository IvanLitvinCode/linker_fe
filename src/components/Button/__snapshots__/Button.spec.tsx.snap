// Vitest Snapshot v1

exports[`Button > default button 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 0px;
}

.c2 {
  font-size: inherit;
  font-family: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --tw-ring-color: var(--emerald-900);
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0,0,0.2,1);
  transition-timing-function: cubic-bezier(0,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,1,1);
  transition-timing-function: cubic-bezier(0.4,0,1,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  border-radius: 0.25rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 700;
  border-width: 1px;
  border-width: 1px;
  border-color: var(--gray-900);
  color: var(--gray-900);
}

.c0:focus-visible {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000);
}

.c0:disabled {
  cursor: not-allowed;
}

.c0:focus-visible {
  border-color: var(--gray-200);
  background-color: var(--gray-200);
}

.c0:active {
  background-color: var(--gray-300);
}

@media (hover:hover) and (pointer:fine) {
  .c0:hover {
    background-color: var(--gray-200);
  }
}

<button
  class="c0"
  type="button"
>
  <span
    class="c1"
    data-label-container="true"
  >
    <span
      class="c2"
      data-label="true"
    >
      Button
    </span>
  </span>
</button>
`;

exports[`Button > default full-width button 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 0px;
}

.c2 {
  font-size: inherit;
  font-family: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --tw-ring-color: var(--emerald-900);
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0,0,0.2,1);
  transition-timing-function: cubic-bezier(0,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  width: 100%;
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,1,1);
  transition-timing-function: cubic-bezier(0.4,0,1,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  border-radius: 0.25rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 700;
  border-width: 1px;
  border-width: 1px;
  border-color: var(--gray-900);
  color: var(--gray-900);
}

.c0:focus-visible {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000);
}

.c0:disabled {
  cursor: not-allowed;
}

.c0:focus-visible {
  border-color: var(--gray-200);
  background-color: var(--gray-200);
}

.c0:active {
  background-color: var(--gray-300);
}

@media (hover:hover) and (pointer:fine) {
  .c0:hover {
    background-color: var(--gray-200);
  }
}

<button
  class="c0"
  type="button"
>
  <span
    class="c1"
    data-label-container="true"
  >
    <span
      class="c2"
      data-label="true"
    >
      Button
    </span>
  </span>
</button>
`;

exports[`Button > default link button 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 0px;
}

.c2 {
  font-size: inherit;
  font-family: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --tw-ring-color: var(--emerald-900);
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0,0,0.2,1);
  transition-timing-function: cubic-bezier(0,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,1,1);
  transition-timing-function: cubic-bezier(0.4,0,1,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  border-radius: 0.25rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 700;
  color: var(--emerald-900);
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
}

.c0:focus-visible {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000);
}

.c0:disabled {
  cursor: not-allowed;
}

.c0:active {
  border-width: 1px;
  border-color: var(--emerald-900);
  --tw-bg-opacity: 1;
  background-color: rgb(255 255 255 / var(--tw-bg-opacity));
}

.c0:disabled {
  color: var(--gray-300);
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
}

@media (hover:hover) and (pointer:fine) {
  .c0:hover {
    background-color: var(--emerald-100);
  }
}

<button
  class="c0"
  type="button"
>
  <span
    class="c1"
    data-label-container="true"
  >
    <span
      class="c2"
      data-label="true"
    >
      Button
    </span>
  </span>
</button>
`;

exports[`Button > default primary button 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 0px;
}

.c2 {
  font-size: inherit;
  font-family: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --tw-ring-color: var(--emerald-900);
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0,0,0.2,1);
  transition-timing-function: cubic-bezier(0,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  border-radius: 0.25rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 700;
  --tw-text-opacity: 1;
  color: rgb(255 255 255 / var(--tw-text-opacity));
  background-color: var(--emerald-900);
}

.c0:focus-visible {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000);
}

.c0:disabled {
  cursor: not-allowed;
}

.c0:focus-visible {
  background-color: var(--emerald-900);
}

.c0:active {
  background-color: var(--emerald-900);
}

@media (hover:hover) and (pointer:fine) {
  .c0:hover {
    background-color: var(--emerald-800);
  }
}

<button
  class="c0"
  type="button"
>
  <span
    class="c1"
    data-label-container="true"
  >
    <span
      class="c2"
      data-label="true"
    >
      Button
    </span>
  </span>
</button>
`;

exports[`Button > default text button 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 0px;
}

.c2 {
  font-size: inherit;
  font-family: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --tw-ring-color: var(--emerald-900);
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0,0,0.2,1);
  transition-timing-function: cubic-bezier(0,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,1,1);
  transition-timing-function: cubic-bezier(0.4,0,1,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  border-radius: 0.25rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 700;
  border-width: 1px;
  border-width: 1px;
  border-color: var(--gray-300);
  --tw-bg-opacity: 1;
  background-color: rgb(255 255 255 / var(--tw-bg-opacity));
  color: var(--gray-900);
}

.c0:focus-visible {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000);
}

.c0:disabled {
  cursor: not-allowed;
}

.c0:focus-visible {
  border-color: var(--gray-200);
  background-color: var(--gray-200);
}

.c0:active {
  background-color: var(--gray-300);
}

@media (hover:hover) and (pointer:fine) {
  .c0:hover {
    border-color: var(--gray-300);
    background-color: var(--gray-100);
  }
}

<button
  class="c0"
  type="button"
>
  <span
    class="c1"
    data-label-container="true"
  >
    <span
      class="c2"
      data-label="true"
    >
      Button
    </span>
  </span>
</button>
`;

exports[`Button > oauth button 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-left: 0.5rem;
}

.c4 {
  font-size: inherit;
  font-family: inherit;
}

.c1 svg,
.c1 img {
  display: block;
  width: 16px;
  height: 16px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-left: 0px;
  margin-right: 0.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --tw-ring-color: var(--emerald-900);
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0,0,0.2,1);
  transition-timing-function: cubic-bezier(0,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,1,1);
  transition-timing-function: cubic-bezier(0.4,0,1,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  border-radius: 0.25rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 700;
  border-width: 1px;
  border-color: var(--gray-700);
  color: var(--gray-700);
  font-size: 1.125rem;
  line-height: 1.5rem;
  font-weight: 700;
}

.c0:focus-visible {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000);
}

.c0:disabled {
  cursor: not-allowed;
}

<button
  class="c0"
  type="button"
>
  <span
    class="c1 c2"
    data-start-icon-container="true"
  >
    <i
      class="neu-share"
    />
  </span>
  <span
    class="c3"
    data-label-container="true"
  >
    <span
      class="c4"
      data-label="true"
    >
      Button
    </span>
  </span>
</button>
`;

exports[`Button > primary and disabled button 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 0px;
}

.c2 {
  font-size: inherit;
  font-family: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --tw-ring-color: var(--emerald-900);
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0,0,0.2,1);
  transition-timing-function: cubic-bezier(0,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  border-radius: 0.25rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 700;
  --tw-text-opacity: 1;
  color: rgb(255 255 255 / var(--tw-text-opacity));
  background-color: var(--gray-300);
}

.c0:focus-visible {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000);
}

.c0:disabled {
  cursor: not-allowed;
}

.c0 svg {
  fill: currentColor;
  color: var(--gray-400);
}

<button
  class="c0"
  disabled=""
  type="button"
>
  <span
    class="c1"
    data-label-container="true"
  >
    <span
      class="c2"
      data-label="true"
    >
      Button
    </span>
  </span>
</button>
`;

exports[`Button > primary and large button 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 0px;
}

.c2 {
  font-size: inherit;
  font-family: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --tw-ring-color: var(--emerald-900);
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0,0,0.2,1);
  transition-timing-function: cubic-bezier(0,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  border-radius: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 1.125rem;
  line-height: 1.5rem;
  font-weight: 900;
  --tw-text-opacity: 1;
  color: rgb(255 255 255 / var(--tw-text-opacity));
  background-color: var(--emerald-900);
}

.c0:focus-visible {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000);
}

.c0:disabled {
  cursor: not-allowed;
}

.c0:focus-visible {
  background-color: var(--emerald-900);
}

.c0:active {
  background-color: var(--emerald-900);
}

@media (hover:hover) and (pointer:fine) {
  .c0:hover {
    background-color: var(--emerald-800);
  }
}

<button
  class="c0"
  type="button"
>
  <span
    class="c1"
    data-label-container="true"
  >
    <span
      class="c2"
      data-label="true"
    >
      Button
    </span>
  </span>
</button>
`;

exports[`Button > primary and small button 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 0px;
}

.c2 {
  font-size: inherit;
  font-family: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --tw-ring-color: var(--emerald-900);
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0,0,0.2,1);
  transition-timing-function: cubic-bezier(0,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  border-radius: 0.25rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 400;
  --tw-text-opacity: 1;
  color: rgb(255 255 255 / var(--tw-text-opacity));
  background-color: var(--emerald-900);
}

.c0:focus-visible {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000);
}

.c0:disabled {
  cursor: not-allowed;
}

.c0:focus-visible {
  background-color: var(--emerald-900);
}

.c0:active {
  background-color: var(--emerald-900);
}

@media (hover:hover) and (pointer:fine) {
  .c0:hover {
    background-color: var(--emerald-800);
  }
}

<button
  class="c0"
  type="button"
>
  <span
    class="c1"
    data-label-container="true"
  >
    <span
      class="c2"
      data-label="true"
    >
      Button
    </span>
  </span>
</button>
`;

exports[`Button > text button with background 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 0px;
}

.c2 {
  font-size: inherit;
  font-family: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --tw-ring-color: var(--emerald-900);
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0,0,0.2,1);
  transition-timing-function: cubic-bezier(0,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  transition-property: color,background-color,border-color,text-decoration-color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,1,1);
  transition-timing-function: cubic-bezier(0.4,0,1,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  border-radius: 0.25rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 700;
  border-width: 1px;
  border-width: 1px;
  border-color: var(--gray-300);
  --tw-bg-opacity: 1;
  background-color: rgb(255 255 255 / var(--tw-bg-opacity));
  color: var(--gray-900);
}

.c0:focus-visible {
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000);
}

.c0:disabled {
  cursor: not-allowed;
}

.c0:focus-visible {
  border-color: var(--gray-200);
  background-color: var(--gray-200);
}

.c0:active {
  background-color: var(--gray-300);
}

@media (hover:hover) and (pointer:fine) {
  .c0:hover {
    border-color: var(--gray-300);
    background-color: var(--gray-100);
  }
}

<button
  class="c0 bg-white"
  type="button"
>
  <span
    class="c1"
    data-label-container="true"
  >
    <span
      class="c2"
      data-label="true"
    >
      Button
    </span>
  </span>
</button>
`;
